src/operations/monochrome.rs

```diff
@@ -0,0 +1,94 @@
+use crate::{error::MagickError, image::Image, wm_err};
+use image::{DynamicImage, GrayImage, Luma};
+
+pub fn monochrome(image: &mut Image) -> Result<(), MagickError> {
+    let mut grayscale = image.pixels.to_luma8();
+    apply_contrast(&mut grayscale, CONTRAST_FACTOR);
+    let noise_texture = NoiseTexture::load()?;
+    apply_dithering(&mut grayscale, &noise_texture);
+    image.pixels = DynamicImage::ImageLuma8(grayscale);
+
+    Ok(())
+}
+
+/// This is empirically tuned to give results similar to ImageMagick's -monochrome
+const CONTRAST_FACTOR: f32 = 10.0;
+
+fn apply_contrast(image: &mut GrayImage, contrast_factor: f32) {
+    let offset = 128.0 * (1.0 - contrast_factor);
+
+    for pixel in image.pixels_mut() {
+        for channel in pixel.0.iter_mut() {
+            let value = *channel as f32;
+            let adjusted = (value * contrast_factor + offset).clamp(0.0, 255.0);
+            *channel = adjusted as u8;
+        }
+    }
+}
+
+const BACKGROUND: Luma<u8> = Luma([255]);
+const FOREGROUND: Luma<u8> = Luma([0]);
+const DITHER_THRESHOLD: u8 = 60;
+
+fn apply_dithering(image: &mut GrayImage, noise_texture: &NoiseTexture) {
+    let width = image.width();
+    let height = image.height();
+
+    for y in 0..height {
+        for x in 0..width {
+            let pixel_luma = image.get_pixel(x, y).0[0];
+
+            if pixel_luma >= (255 - DITHER_THRESHOLD) || pixel_luma <= (0 + DITHER_THRESHOLD) {
```

Comment:
We don't need a dither threshold at all, the contrast step already takes care of the "thresholding" by amplifying the pixel values and pushing them all the way to 0 or 255.

With `DITHER_THRESHOLD` of 0 and contrast `CONTRAST_FACTOR` of 2.0 this is what I am getting:

<img width="500" height="30" alt="out-wm" src="https://github.com/user-attachments/assets/af4df733-126e-4828-9bea-04c7df3ad3b7" />

Imagemagick reference:

<img width="500" height="30" alt="out-im" src="https://github.com/user-attachments/assets/dd4f2f5a-abd5-45dc-a981-ac5e12c26a86" />